velox/functions/prestosql/ArithmeticImpl.h

```diff
@@ -161,4 +161,49 @@ FOLLY_ALWAYS_INLINE double truncate(
   }
 }
 
+// helper function for calculating upper and lower limit of wilson interval
+template <bool upper>
+FOLLY_ALWAYS_INLINE double
+wilsonInterval(int64_t successes, int64_t trials, double z) {
+  VELOX_USER_CHECK_GE(successes, 0, "number of successes must not be negative");
+  VELOX_USER_CHECK_GT(trials, 0, "number of trials must be positive");
+  VELOX_USER_CHECK_LE(
+      successes,
+      trials,
+      "number of successes must not be larger than number of trials");
+  VELOX_USER_CHECK_GE(z, 0, "z-score must not be negative");
+
+  double s{static_cast<double>(successes)};
+  double n{static_cast<double>(trials)};
+  double p{s / n};
+
+  // Wilson interval limits are solutions of a quadratic equation.
+  // Let the equation be {ax^2 + bx + c = 0}.
+  // r will store the value (-b + sqrt(b*b - 4*a*c)).
+  double a, c, r;
+
+  // Compute the equations differently depending on whether z is large or small.
+  // This helps to avoid computations like (INFINITY/INFINITY),
+  // yielding accurate results in the limit as z approaches infinity.
+  if (z < 1) {
+    if (upper)
+      a = n + z * z;
```

Comment:
Please wrap the statement into curly braces (block) for the if-else where applicable i.e. 
```
if (upper) {
   a = n + z * z;
} else {
   c = s * p;
}

Reply:
I've changed the `if` condition for the return value to use curly braces.

The other two `if` conditions might not be needed at all. Since the computation of `result` is based on a compile-time constant, the compiler should recognise that either `a` or `c` is unused and optimise it out. Removing the `if` condition there would not affect performance at all but will make the code more readable. Does this sound reasonable?

Reply:
Thanks! Yes, you'd think that it could. Probably have to look at the generated assembler to see what it is doing for the two versions of the function to see if it optimized away one of the not used variables.